src/vm_cpu.rs

```diff
@@ -188,6 +189,24 @@ impl Iterator for CounterMaskIter {
 }
 
 impl VmPmuState {
+    // Sets the bit to enable access to the CSR for counter_index
+    fn set_hcounteren_value_bit(counter_index: LogicalCounterIndex) {
+        // Unwrap ok: Guaranteed to succeed since we have already tested the condition in the call chain
+        let pmu_info = pmu::PmuInfo::get().unwrap();
+        let csr = pmu_info.get_csr_for_counter(counter_index).unwrap();
+        CSR.hcounteren
+            .set(CSR.hcounteren.get() | 1 << (csr - CSR_CYCLE as u64));
```

Comment:
Setting an individual bit can be done with a single instruction using `CSR.hcounteren.read_and_set_bit()`